File: Baekjoon_Online_Judge/13_geometry_1/05_boj_1002.cpp
#include <iostream>
#include <cmath>

int intersection(int x1, int y1, int r1, int x2, int y2, int r2){
	// 두 원이 일치하는 경우 교점 무한개
	if(x1 == x2 and y1 == y2 and r1 == r2)
    	return -1;
    // 두 원의 중심이 같고 반지름이 다른 경우 교점 0개
    if (x1 == x2 and y1 == y2 and r1 != r2)
    	return 0;
    double distance = sqrt(pow(abs(x2-x1), 2) + pow(abs(y2-y1), 2));
    // 한 원이 나머지 한 원에 포함되는 경우 교점 0개
    if(distance < abs(r1-r2))
    	return 0;
    // 두 원 사이의 거리 > 반지름 합인 경우 교점 0개 
    if(distance > r1+r2)
    	return 0;
    // 두 원이 안에서 접하는 경우 교점 1개
    if(distance == abs(r1-r2))
    	return 1;
    // 두 원이 밖에서 접하는 경우 교점 1개
    if(distance == r1+r2)
    	return 1;
    // 두 원 사이의 거리 < 반지름 합이면서 한 원이 다른 원에 포함이 아닌 경우 교점 2개
    if(distance < r1+r2 and distance > abs(r1-r2))
    	return 2;
}

int main(){
    int x1, y1, r1, x2, y2, r2, n;
    scanf("%d", &n);
    for(int i = 0; i<n; i++){
    	scanf("%d %d %d %d %d %d", &x1, &y1, &r1, &x2, &y2, &r2);
    	printf("%d\n", intersection(x1, y1, r1, x2, y2, r2));
    }
}

File: Baekjoon_Online_Judge/13_geometry_1/05_boj_1002.js
const fs = require("fs");
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

input.shift();
input = input.map(v => v.split(" ").map(v => parseInt(v)));
const result = input.map(([x1, y1, r1, x2, y2, r2]) => {
	if(x1 === x2 && y1 === y2){
		// 일치
		if(r1 === r2){
			return -1;
		}
		// 포함
		else{
			return 0;
		}
	}
	const distance = Math.sqrt(Math.abs(x1 - x2)**2 + Math.abs(y1 - y2)**2);
	const sum = r1 + r2;
	const diff = Math.abs(r1 - r2);
	// 외접, 내접
	if(distance === sum || distance === diff){
		return 1;
	}
	// 두 점에서 만남
	else if(distance < sum && distance > diff){
		return 2;
	}
	// 밖에서 안 만남, 안에서 안 만남
	else{
		return 0;
	}
});

console.log(result.join("\n"));